service/worker/deployment/deployment_util.go

```diff
@@ -133,32 +134,97 @@ func StartAndUpdateDeploymentWorkflow(
 						Request:     updateReq,
 					},
 				},
 			},
 		},
 	})
 
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// GenerateDeploymentWorkflowID is a helper that generates a system accepted
+// workflowID which are used in our deployment workflows
+func (d *DeploymentWorkflowClient) GenerateDeploymentWorkflowID() string {
+	return DeploymentWorkflowIDPrefix + DeploymentWorkflowIDDelimeter + d.deployment.DeploymentName + DeploymentWorkflowIDDelimeter + d.deployment.BuildId
+}
+
+// GenerateStartWorkflowPayload generates start workflow execution payload
+func (d *DeploymentWorkflowClient) GenerateStartWorkflowPayload() (*commonpb.Payloads, error) {
+	workflowArgs := &deployspb.DeploymentWorkflowArgs{
+		NamespaceName:     d.namespace.Name().String(),
+		NamespaceId:       d.namespace.ID().String(),
+		DeploymentName:    d.deployment.DeploymentName,
+		BuildId:           d.deployment.BuildId,
+		TaskQueueFamilies: nil,
+	}
+	workflowInputPayloads, err := sdk.PreferProtoDataConverter.ToPayloads(workflowArgs)
+	if err != nil {
+		return nil, err
+	}
+	return workflowInputPayloads, nil
+}
+
+// GenerateUpdateDeploymentPayload generates update workflow payload
+func (d *DeploymentWorkflowClient) GenerateUpdateDeploymentPayload(taskQueueName string, taskQueueType enumspb.TaskQueueType,
+	pollTimestamp *timestamppb.Timestamp) (*commonpb.Payloads, error) {
+	updateArgs := &deployspb.RegisterWorkerInDeploymentArgs{
+		TaskQueueName:   taskQueueName,
+		TaskQueueType:   taskQueueType,
+		FirstPollerTime: nil, // TODO Shivam - come back to this
+	}
+	updatePayload, err := sdk.PreferProtoDataConverter.ToPayloads(updateArgs)
 	if err != nil {
 		return nil, err
 	}
+	return updatePayload, nil
+}
+
+// ValidateDeploymentWfParams is a helper that verifies if the fields used for generating
+// deployment related workflowID's are valid
+func (d *DeploymentWorkflowClient) ValidateDeploymentWfParams(fieldName string, field string, maxIDLengthLimit int) error {
+	// Length checks
+	if field == "" {
+		return serviceerror.NewInvalidArgument(fmt.Sprintf("%v cannot be empty", fieldName))
+	}
+
+	// Length of each field should be: (MaxIDLengthLimit - prefix and delimeter length) / 2
+	if len(field) > (maxIDLengthLimit-DeploymentWorkflowIDInitialSize)/2 {
+		return serviceerror.NewInvalidArgument(fmt.Sprintf("size of %v larger than the maximum allowed", fieldName))
+	}
+
+	// Invalid character check - Cannot contain the reserved delimeter ("|") since its being used for building deployment workflowID
+	if strings.Contains(field, DeploymentWorkflowIDDelimeter) {
+		return serviceerror.NewInvalidArgument(fmt.Sprintf("%v cannot contain reserved prefix %v", fieldName, DeploymentWorkflowIDDelimeter))
+	}
+
+	// UTF-8 check
+	return common.ValidateUTF8String(fieldName, field)
+}
 
-	return resp, nil
+func (d *DeploymentWorkflowClient) EscapeChar(s string) string {
+	s = strings.Replace(s, `\`, `\\`, -1)
+	s = strings.Replace(s, DeploymentWorkflowIDDelimeter, `\`, -1)
```

Comment:
```suggestion
s = strings.Replace(s, DeploymentWorkflowIDDelimeter, `\`+DeploymentWorkflowIDDelimeter, -1)
```
i.e, turn `abc|def\ghi` into `abc\|def\\ghi`. Otherwise `abc\def` and `abc||def` will escape to the same thing. We need this function to be injective

Reply:
playing with strings, in this manner, is often so confusing. Thanks for the catch + idea!